lib/_http_client.js

```diff
@@ -591,11 +591,12 @@ function responseKeepAlive(res, req) {
 
   if (!req.shouldKeepAlive) {
     if (socket.writable) {
-      debug('AGENT socket.destroySoon()');
-      if (typeof socket.destroySoon === 'function')
-        socket.destroySoon();
+      socket.end();
+
+      if (socket.writableFinished)
+        socket.destroy();
       else
-        socket.end();
+        socket.once('finish', socket.destroy);
```

Comment:
Do we even need these calls to `destroy()` doesn't `autoDestroy` already handle this?